Structure_Engine/Query/VolumetricMaterialTakeoff.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BH.oM.Dimensional;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.oM.Base.Attributes;
+using BH.oM.Physical.Materials;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.MaterialFragments;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the volumetric material takeoff from the PileFoundation object. The takeoff will contain materials and volumes from the PileCap and Piles.")]
+        [Input("pileFoundation", "The PileFoundation object to extract the volumetric material takeoff from.")]
+        [Output("volTakeoff", "The volumetric material takeoff based on buildup of the PileFoundation object.")]
+        public static VolumetricMaterialTakeoff VolumetricMaterialTakeoff(this PileFoundation pileFoundation)
```

Comment:
Are you sure we want/need the material takeoffs for analytical elements? My question originates from the fact that for some reason we do not have them for bars and panels, curious if it was not intended to for example push the users to use `Physical` objects for material purposes. I am 100% ignorant in the topic, so please feel free to ignore 👍

Reply:
The `VolumetricMaterialTakeoff` does work for `Bar` and `Panel` though  - I'd been keen to have them included for `Structure.Elements` in general so material take offs can be done when optioneering different structural typologies.

Reply:
Right, I got confused by the fact that there had not been any `VolumetricMaterialTakeoff` method in Structure_Engine. But following that fact, if `Bar` works with a dedicated overload, why would you need it for `Pile`?

Reply:
Remove `VolumetricMaterialTakeoff` and add `SolidVolume` method for `Pile`/`PileFoundation` and `MaterialComposition` for `Pile`/`PileFoundation`. 

The `PadFoundation` are already covered for `IAreaElement` which `PadFoundation` implements.